homeassistant/components/palazzetti/coordinator.py

```diff
@@ -0,0 +1,76 @@
+"""Helpers to help coordinate updates."""
+
+from typing import TypedDict
+
+from pypalazzetti.client import PalazzettiClient
+from pypalazzetti.exceptions import CommunicationError, ValidationError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    AVAILABLE,
+    DOMAIN,
+    EXHAUST_TEMPERATURE,
+    FAN_SPEED,
+    IS_HEATING,
+    LOGGER,
+    OUTLET_TEMPERATURE,
+    PELLET_QUANTITY,
+    ROOM_TEMPERATURE,
+    SCAN_INTERVAL,
+    TARGET_TEMPERATURE,
+)
+
+
+class PalazzettiData(TypedDict):
+    """Class for defining data in dict."""
+
+    available: bool
+    is_heating: bool
+    target_temperature: int
+    room_temperature: float
+    outlet_temperature: float
+    exhaust_temperature: float
+    pellet_quantity: int
+    fan_speed: int
+
+
+class PalazzettiDataUpdateCoordinator(DataUpdateCoordinator[PalazzettiData]):
+    """Class to manage fetching Palazzetti data from a Palazzetti hub."""
+
+    entry: ConfigEntry
+    palazzetti: PalazzettiClient
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize global Palazzetti data updater."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+        self.entry = entry
+        self.palazzetti = PalazzettiClient(entry.data[CONF_HOST])
```

Comment:
```suggestion
    def __init__(self, hass: HomeAssistant) -> None:
        """Initialize global Palazzetti data updater."""
        super().__init__(
            hass,
            LOGGER,
            name=DOMAIN,
            update_interval=SCAN_INTERVAL,
        )
        self.palazzetti = PalazzettiClient(self.config_entry.data[CONF_HOST])
```

Reply:
I need the `ConfigEntry` downstream in climate.py `__init__` when using the MAC address as an identifier in the `DeviceInfo`

Reply:
I'm not following? You can just access it via `coordinator.config_entry`

Reply:
I am confused because your suggestion is removing `confg_entry` entirely from the `__init__` signature and is not passing it to the `super.__init` call. I just had a look at `homeassistant.helpers.update_coordinator.DataUpdateCoordinator.__init__` and it has this comment which suggests it should be passed:
````
        if config_entry is UNDEFINED:
            self.config_entry = config_entries.current_entry.get()
            # This should be deprecated once all core integrations are updated
            # to pass in the config entry explicitly.
        else:
            self.config_entry = config_entry
````